eclair-core/src/main/scala/fr/acinq/eclair/message/OnionMessages.scala

```diff
@@ -43,6 +51,18 @@ object OnionMessages {
   case class Recipient(nodeId: PublicKey, pathId: Option[ByteVector], padding: Option[ByteVector] = None, customTlvs: Set[GenericTlv] = Set.empty) extends Destination
   // @formatter:on
 
+  // @formatter:off
+  sealed trait RoutingStrategy
+  object RoutingStrategy {
+    /** Use the provided route to reach the recipient or the blinded path's introduction node. */
+    case class UseRoute(intermediateNodes: Seq[PublicKey]) extends RoutingStrategy
+    /** Directly connect to the recipient or the blinded path's introduction node. */
+    val connectDirectly: UseRoute = UseRoute(Nil)
```

Comment:
My confusion came from the fact that I assumed that the route provided in `UseRoute` had to start with a node with whom we have channels. In that case, there would be a discrepancy because in one case we would assume that we're already connected, and in the other we would have to actively make try to connect. But this premise was actually wrong, users are completely free to provide a predetermined route through a node that we don't have channels with, and in that case the way you're handling it is perfectly consistent!